### PR TITLE
Callback function that returns host array --- another method

### PR DESCRIPTION
**Pull request by nychiang**
Address issue #533 
Adding a new user option to determine to which `mem_space` should HiOp return the solutions.

### CODE REVIEW

**Comment by maksud** [comment]
Line 1085 of file `src/Optimization/hiopNlpFormulation.cpp` in HiOp you have 
```
if(options->GetString("callback_mem_space")=="device") {
```

It should be 
```
if(options->GetString("callback_mem_space")=="host") {
```

File: src/Optimization/hiopNlpFormulation.cpp [hunk 1]
```diff
@@ -1082,16 +1082,32 @@ void hiopNlpFormulation::user_callback_solution(hiopSolveStatus status,
...
-                                   (int)n_vars_,
-                                   x.local_data_const(),
-                                   z_L.local_data_const(),
-                                   z_U.local_data_const(),
-                                   (int)n_cons_,
-                                   cons_body_->local_data_const(),
-                                   cons_lambdas_->local_data_const(),
-                                   inv_obj_scale*obj_value);
-                                   
+  if(options->GetString("callback_mem_space")=="device") {
```
Comment (cnpetra) [inline src/Optimization/hiopNlpFormulation.cpp:1085]: see @maksud comment

**Review by cnpetra — CHANGES_REQUESTED** [review]

File: src/Optimization/hiopNlpFormulation.cpp (hunk 1 above)
Comment (nychiang) [reply src/Optimization/hiopNlpFormulation.cpp:1085]: umm... it's github's problem. I did fix this issue in the last commit [9e8c8ab](https://github.com/LLNL/hiop/pull/535/commits/9e8c8ab1c64a04787519f6ad3ebbfb8f006e3ec6), but it seems that github didn't show it.
@cnpetra @maksud

Comment (nychiang) [reply src/Optimization/hiopNlpFormulation.cpp]: just push the change

**Review by cnpetra — APPROVED** [review]

**Pull request merged**